src/baldr/datetime.cc

```diff
@@ -13,28 +13,59 @@
 #include "midgard/logging.h"
 #include "midgard/util.h"
 
+namespace {
+// use a cache to store already constructed sys_info's since they aren't cheap
+template <typename TP>
+const date::sys_info&
+from_cache(const TP& tp,
+           const date::time_zone* tz,
+           std::unordered_map<const date::time_zone*, std::vector<date::sys_info>>& cache) {
+  // check if we have anything for this timezone in the cache
+  auto tz_it = cache.find(tz);
+  if (tz_it != cache.cend()) {
+    // we have something in the cache, see if one of the infos has the particular time point in range
+    auto st = date::floor<std::chrono::seconds>(tp.get_sys_time());
+    auto info_it =
+        std::find_if(tz_it->second.begin(), tz_it->second.end(),
+                     [&st](const date::sys_info& info) { return info.begin <= st && st < info.end; });
```

Comment:
I assume the array in the map is usually 1-2 elements long? There wouldn't be any point in sorting it according to begin time which would allow a binary search?

Reply:
hahah i had that as a TODO but  you are exactly right. there is basically no benefit. the only time the array isnt a single entry long is if the route crosses over daylight savings time